Simplenote/PDFLinkActivityItemProvider.swift

```diff
@@ -0,0 +1,24 @@
+import Foundation
+
+public class PDFLinkActivityItemProvider : UIActivityItemProvider {
+    public let targetURL: URL
+    public let filename: String
+    public let note: Note
+    
+    public init(note: Note) {
+        self.note = note
+        self.filename = String(format: "%@.pdf", "Thisisastring")
+        self.targetURL = FileManager.documentsURL.appendingPathComponent(filename)
+        
+        super.init(placeholderItem: targetURL)
+    }
+    
+    public override var item: Any {
+        guard activityType?.rawValue != "com.apple.UIKit.activity.RemoteOpenInApplication",
```

Comment:
Let's move the inline string to a constant 😄 

You can do something like...

```
private enum Settings {
    static let supportedActivityType = "..."
}
```

(Or a simple `let` constant on top of the file with a comment would just be ferpect!)

Reply:
**PLUS** are we 100% sure we need to check on the Activity Type?

Since we're exporting a PDF and passing along a URL that contains the `.pdf` extension, perhaps we can drop the `activityType` check?

Furthermore, probably checking on the activity type on a parameter-less getter is probably not good. In theory, getters should behave the same, regardless of the caller. Otherwise, the internal behavior would be hard to tell, from outside the API.

Reply:
I didn't think that we did, but if it is not there then the share for things like mail and apple notes were getting the note content and a PDF.  By adding this check in before creating the PDF we can stop that